packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -274,10 +284,13 @@ export const makeVaultManager = (
     });
   };
 
-  /** @type {VaultManager} */
   return Far('vault manager', {
     ...shared,
     makeLoanKit,
     liquidateAll,
   });
 };
+
+/**
+ * @typedef {ReturnType<makeVaultManager>} VaultManager
```

Comment:
this might be a useful pattern, that a module that exports `makeFooThing` also exports a `FooThing` type alias for `ReturnType<makeFooThing>`.

Reply:
Sure but I'm a little wary of relying on the type inference here, which isn't visible in code diffs. I'd much prefer have `VaultManager` be an explicitly defined type, and have `makeVaultManager` have an `@returns {VaultManager}`, so that if the implementation deviates, the implementation or type has to be explicitly updated.

Reply:
I see where you're coming from and agree for most functions. Updating it in two places is extra work but it has a benefit for most functions to detect if the implementation fails to satisfy the declared signature of the function. 

But in this case, I think the implementation declares the API, the way a class definition does. You don't define the class return type and the class implementation separately. These `makeFooThing` are serving like constructors for `FooThing`. If what `makeFooThing` returns changes, that means the type of `FooThing` is changing.

That said, a function has more freedom in what it returns, but I think on balance we can proceed this way and fix any problems that pop up. There's also the pragmatic consideration: the pain point right now is navigation to these faux-constructors and we can solve that most quickly this way.